packages/web-components/src/components/masthead/left-nav.ts

```diff
@@ -136,6 +136,12 @@ class DDSLeftNav extends StableSelectorMixin(BXSideNav) {
       if (expanded) {
         this._hFocusWrap = focuswrap(this.shadowRoot!, [startSentinelNode, endSentinelNode]);
         doc.body.style.overflow = `hidden`;
+
+        // TODO: remove this logic once masthead can account for banners.
+        // calculate top offset to display left-nav correctly when page has banner above masthead
+        const masthead: HTMLElement | null = doc?.querySelector('dds-masthead');
+        const mastheadTopOffset = masthead?.offsetTop;
+        this.style.top = `${mastheadTopOffset}px`;
```

Comment:
I think this top calculation works great on page load, however, I think we need to bind this to a window resize event when users decided to click on the "Stay here" button, which closes the locale banner, thus the top offset position value will need to be updated.

![Build_and_modernize_-_Singapore___IBM](https://user-images.githubusercontent.com/1815714/134363048-3e3b780d-e5e3-418c-9227-c6a7d3ca3beb.jpg)

Reply:
This is an edge case and might not be needed for this patch.

Reply:
tried using `resizeObserver` to detect body height change when the banner disappears but it doesn't seem to detect the change locally. Hoping that https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7216 will help to incorporate the masthead with some awareness of components above and that it will resolve the edge case that we're seeing here